### PR TITLE
check_compliance.py: Exclude simple Kconfig token pasting

### PR DESCRIPTION
**Pull request by Sizurka**
Alter the regular expression used in git-grep to exclude simple token pasting for the form `CONFIG_STUFF##something`.  This does not catch any use for `UTIL_CAT` and friends, but it does allow removal of some elements from the whitelist.

The specific motivation is zephyrproject-rtos/zephyr#15105 where I used token pasting for counter instances.  However, this is a general solution to the simple cases like that.

### CODE REVIEW

File: scripts/check_compliance.py [hunk 1]
```diff
@@ -314,11 +314,12 @@ def check_no_undef_outside_kconfig(self, kconf):
         # Kconfig files that are not part of the main Kconfig tree, which will
         # trigger false positives until we do something fancier. Skip
         # doc/releases too, which often references removed symbols.
-        grep_cmd = "git grep --only-matching --line-number -I --null " \
-                   "--extended-regexp --word-regexp CONFIG_[A-Z0-9_]+ " \
-                   "-- :!samples :!tests :!doc/releases"
+        grep_cmd = ["git", "grep", "--only-matching", "--line-number", "-I",
+                    "--null", "--perl-regexp",
+                    r"\bCONFIG_[A-Z0-9_]+\b(?!\s*#)",
```
Comment (ulfalizer) [inline scripts/check_compliance.py]: Is there any harm in making this `(?!\s*##)` by the way, just to make it a bit more specific?

The check searches comments and stuff too, where maybe `CONFIG_FOO  #...` could show up in some context, though it's unlikely.

Comment (Sizurka) [reply scripts/check_compliance.py]: Done for both.  I also missed an instance that could be removed from the whitelist the first time through (`CONFIG_UART_`); I only checked for ones that matched the new exclusion (i.e. that where present as simple token pastes), but I managed to miss that one.

Comment (ulfalizer) [reply scripts/check_compliance.py]: LGTM

**Comment by galak** [comment]
While not related to this PR, what to people feel about moving the whitelist into a file and keeping it in the zephyr repo itself.

**Comment by ulfalizer** [comment]
@galak 
The reason I didn't do that right away is that I think the right solution is to move `ci-tools` back into the main Zephyr repo. There will be other whitelists later on too, and it's annoying if you have to have a custom whitelist file for each, that supports comments, etc., and with the extra code that comes with that.

It feels like working around that `ci-tools` would be better to have as `scripts/ci` in the Zephyr repo, where it used to be.

I was kinda hoping that someone would take the initiative and move it back, but I could make a PR for it later.

**Comment by ulfalizer** [comment]
@Sizurka 
Spottedanother thing: The comment above `UNDEF_KCONFIG_WHITELIST` needs to be updated. Could just say "many of these are symbols used as example" now.

**Comment by Sizurka** [comment]
Ok, comment updated.

**Pull request merged**